mentions/src/main/kotlin/io/spine/examples/pingh/mentions/GitHubClientProcess.kt

```diff
@@ -80,20 +96,94 @@ public class GitHubClientProcess :
                 .build()
         }
 
+        // TODO:2024-05-21:mykyta.pimonov: Check that a minute has passed since
+        //  the previous successfully accepted request.
+
+        // TODO:2024-05-20:mykyta.pimonov: Check that the token is not expired.
+
         builder().setWhenStarted(currentTime())
         return MentionsUpdateFromGitHubRequested.newBuilder()
             .setId(command.id)
-            .setToken(command.token)
             .vBuild()
     }
 
+    /**
+     * Updates user mentions by emitting [UserMentioned] events
+     * and terminates the mention update process.
+     *
+     * @return List of events, where the [UserMentioned] event for each mention comes first,
+     * followed by a single [MentionsUpdateFromGitHubCompleted] event.
+     */
     @React
-    internal fun on(event: MentionsUpdateFromGitHubRequested): MentionsUpdateFromGitHubCompleted {
+    internal fun on(event: MentionsUpdateFromGitHubRequested): List<EventMessage> {
         archived = true
-
+        val username = event.id.username
+        val token = builder().token
+        val mentions = gitHubClientService.fetchMentions(username, token)
+        val emittingEvents = createUserMentionedEvents(mentions)
+        emittingEvents.add(
+            MentionsUpdateFromGitHubCompleted.newBuilder()
+                .setId(event.id)
+                .vBuild()
+        )
         builder().clearWhenStarted()
-        return MentionsUpdateFromGitHubCompleted.newBuilder()
-            .setId(event.id)
-            .vBuild()
+        return emittingEvents.toList()
+    }
+
+    private fun createUserMentionedEvents(gitHubMentions: List<GitHubMention>):
+            MutableList<EventMessage> =
```

Comment:
Please stick to immutable structures as much as possible. Here you transform one immutable list into another. And you don't know that you will need a _mutable_ structure elsewhere.